### ISSUE TITLE
[BUG] GetConsumptionUsageDetailsAsync : An operation on a socket could not be performed because the system lacked sufficient buffer space or because a queue was full.

### ISSUE THREAD

**Issue by NovaWasTaKenn** [Azure/azure-sdk-for-net#37851]
### Library name and version

Azure.ResourceManager.Consumption -Version 1.1.0-beta.1

### Describe the bug

The error appears as an aggregateException when calling ConsumptionExtensions.GetConsumptionUsageDetailsAsync(client, scope, "properties/meterDetails").AsPages().GetAsyncEnumerator().MoveNextAsync()

Here's what it looks like as an aggregateException : 
![image](https://github.com/Azure/azure-sdk-for-net/assets/98171103/6dd64861-227f-4218-a482-fafa2de32353)

Here are the two type of errors contained in the aggregateException, the socket one appears 4 times : 
![image](https://github.com/Azure/azure-sdk-for-net/assets/98171103/919bd37e-a38f-4108-a6c1-c46df50c5dd8)
![image](https://github.com/Azure/azure-sdk-for-net/assets/98171103/009218a0-5809-458f-a786-355b5faa910f)

This error appears very rarely right now : 2 times out of 2736 MoveNextAsync calls

### Expected behavior

No error, outputs all the lines of data

### Actual behavior

Encounters the error cited in the description, stops sending data as a result

### Reproduction Steps

AsyncPageable<ConsumptionUsageDetail> usage = ConsumptionExtensions.GetConsumptionUsageDetailsAsync(client, scope, "properties/meterDetails");
IAsyncEnumerator<Azure.Page<ConsumptionUsageDetail>> pagesEnumerator = usage.AsPages().GetAsyncEnumerator();

Then I call MoveNextAsync() to move throught the pages. The error appears 2 times out of 2736 MoveNextAsync() calls

### Environment

Azure Function App, Windows 64 bit, runtime version 3 so .NET Core 3.1

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by HarveyLink** [comment]
Hi @NovaWasTaKenn , Thank you for using Azure SDK for .NET.
Could you provide more detials for this issue, like some response log samples (without sensitive private data).
We'll try to repro this issue on ourside.

**Comment by NovaWasTaKenn** [comment]
Hi @HarveyLink, I managed to reproduce the error and log the responses. Here are the ten last responses before the bug. I, unfortunately, do not have the content of each response (containing the nextLink and the values returned) because the json file in wich I stored them ended up broken due to the amount of data.
[responseLog.zip](https://github.com/Azure/azure-sdk-for-net/files/12320990/responseLog.zip)
